### DISCUSSION TITLE
Is there a way to distinct package version based on platform_machine ?

### DISCUSSION THREAD

**Discussion by liran-cohen-hs** [pdm-project/pdm#1793] [Q&A] [answered]
I will start with the need:
our data science team works with `torch` package, version 1.7.1
our local development station are M1 macs so `torch` is incompatible, fixed on newer versions
what I would like to provide in the `pyproject.toml` and `pdm.lock` is the following logic:
if platform_machine != 'arm64'" -> install torch 1.7.1
else if platform_machine == 'arm64'" -> install torch 1.10.1

the basic command of `pdm add "torch==1.7.1; platform_machine != 'arm64'"` start the flow, but once I add 1.10.1 with a different condition it overrides
Tried to add to dev dependencies as well but then PDM says there is a conflict in the lock obviously

any solution to this flow ? 
Thanks !

**Comment by frostming** [comment, marked as answer]
Not now, see the issue #46

**Reply by liran-cohen-hs** [reply]
thanks for the quick response ! is there any workaround at least for local development ?
what I mean is - I want the lock file to have 1.7.1 because that's what we need in production, now I want developers to run `pdm sync/install` locally and have torch 1.10.1, even with another installation command, ideally, without affecting `pyproject.toml` and lock files
because pip is not installed as part of PDM venv it's not an option and I did not see any other option to add a package without affecting the files

**Reply by frostming** [reply]
You can try the new feature enabled by the latest beta release: multiple lockfile. Specify different version of the same package in two groups, can create separate lockfiles for the selected group respectively.

https://pdm.fming.dev/latest/usage/dependency/#specify-the-lockfile-to-use
